lib/internal/webidl.js

```diff
@@ -263,6 +265,74 @@ function type(V) {
   }
 }
 
+// https://webidl.spec.whatwg.org/#js-dictionary
+function createDictionaryConverter(members) {
+  // The spec requires us to operate members of each dictionary in
+  // lexicographical order. We are doing all of these in the outer scope to
+  // reduce the overhead that could happen in the returned function.
+  const sortedMembers = [];
+  for (let i = 0; i < members.length; i++) {
+    const member = members[i];
+    ArrayPrototypePush(sortedMembers, member);
+  }
+
+  ArrayPrototypeSort(sortedMembers, (a, b) => {
+    if (a.key === b.key) {
+      return 0;
+    }
+    return a.key < b.key ? -1 : 1;
+  });
+
+  return function(
+    V,
+    opts = kEmptyObject,
+  ) {
+    const typeV = type(V);
+    if (typeV !== 'Undefined' && typeV !== 'Null' && typeV !== 'Object') {
+      throw makeException(
+        'can not be converted to a dictionary',
+        opts,
+      );
+    }
+
+    const idlDict = { __proto__: null };
+    for (let i = 0; i < sortedMembers.length; i++) {
+      const member = sortedMembers[i];
+      const key = member.key;
+      let jsMemberValue;
+      if (typeV === 'Undefined' || typeV === 'Null') {
+        jsMemberValue = undefined;
+      } else {
+        jsMemberValue = V[key];
+      }
+
+      if (jsMemberValue !== undefined) {
+        const memberContext = opts.context ? `${key} in ${opts.context}` : `${key}`;
+        const converter = member.converter;
+        const idlMemberValue = converter(
+          jsMemberValue,
+          {
+            __proto__: null,
+            prefix: opts.prefix,
+            context: memberContext,
+          },
+        );
+        idlDict[key] = idlMemberValue;
+      } else if (ObjectPrototypeHasOwnProperty(member, 'defaultValue')) {
+        const idlMemberValue = member.defaultValue;
+        idlDict[key] = idlMemberValue;
```

Comment:
having defaultValue be a getter here will be slower than a function. Once changed to a function you can replace the hasOwnProperty with a `typeof defaultValue === 'function'` or `defaultValue != null` which will be significantly faster.

Reply:
Ah, that explains what I saw in undici! Nice one, will update this.